src/lib.rs

```diff
@@ -179,146 +167,24 @@ impl Reaper {
     }
 
     /// Reap zombie processes forever.
-    async fn reap(&'static self, _driver_guard: async_lock::MutexGuard<'_, ()>) -> ! {
-        loop {
-            // Wait for the next SIGCHLD signal.
-            self.pipe.wait().await;
-
-            // Notify all listeners waiting on the SIGCHLD event.
-            self.sigchld.notify(std::usize::MAX);
-
-            // Reap zombie processes, but make sure we don't hold onto the lock for too long!
-            let mut zombies = mem::take(&mut *self.zombies.lock().unwrap());
-            let mut i = 0;
-            'reap_zombies: loop {
-                for _ in 0..50 {
-                    if i >= zombies.len() {
-                        break 'reap_zombies;
-                    }
-
-                    if let Ok(None) = zombies[i].try_wait() {
-                        i += 1;
-                    } else {
-                        zombies.swap_remove(i);
-                    }
-                }
-
-                // Be a good citizen; yield if there are a lot of processes.
-                //
-                // After we yield, check if there are more zombie processes.
-                future::yield_now().await;
-                zombies.append(&mut self.zombies.lock().unwrap());
-            }
-
-            // Put zombie processes back.
-            self.zombies.lock().unwrap().append(&mut zombies);
-        }
+    async fn reap(&'static self, driver_guard: reaper::Lock) -> ! {
```

Comment:
imo it would make sense to inline this function whereever it is used, now that it is a one-liner